OpenSim/Simulation/Model/Appearance.h

```diff
@@ -33,105 +33,160 @@ namespace OpenSim {
 class Body;
 class Model;
 
+/**
+VisualRepresentation is the OpenSim name used across the OpenSim API, it maps to
+SimTK::DecorativeGeometry::Representation.
+*/
+using VisualRepresentation = SimTK::DecorativeGeometry::Representation;
 
-//=============================================================================
-//=============================================================================
 /**
- * A class that holds the Display Attributes (Appearance) of an object displayed 
- * in the OpenSim Visualizer.
- * 
- * BaseAppearance contains Appearance properties that apply to all geometry.
- * Geometry that have a surface so that it can be textured can use the subclass
- * Appearance, while schematic line drawings (e.g. Arrows, Frames) can use 
- * LineAppearance which offers thickness. 
- *
- * TODO: Add Resolution to Appearance from DecorativeGeometry and 
- * utilize LineAppearance for Arrow, Line, and Frame (unused for now)
- *
- * @author Ayman Habib
- * @version 1.0
- */
-class OSIMSIMULATION_API BaseAppearance : public Object {
-    OpenSim_DECLARE_CONCRETE_OBJECT(BaseAppearance, Object);
+SurfaceAppearance class holds the appearance properties of a piece of Geometry 
+displayed in the OpenSim visualizer or GUI as a surface. The properties in this 
+class are specific to geometry that have surfaces so that these surfaces can be 
+textured, or rendered using  a variety of shading models. 
+*/
+
+class OSIMSIMULATION_API SurfaceAppearance : public Object {
+    OpenSim_DECLARE_CONCRETE_OBJECT(SurfaceAppearance, Object);
 public:
     //==========================================================================
     // PROPERTIES
     //==========================================================================
-    OpenSim_DECLARE_PROPERTY(color, SimTK::Vec3,
-        "The color (rgb) used to display the object. ");
-    OpenSim_DECLARE_PROPERTY(opacity, double,
-        "The opacity (0-1) used to display the object. ");
     OpenSim_DECLARE_PROPERTY(representation, int,
-        "The representation (0:Hidden, 1:Points, 2:Wire 3:Shaded) used to display the object. ");
+     "The representation (1:Points, 2:Wire, 3:Shaded) used to display the object.");
+    OpenSim_DECLARE_OPTIONAL_PROPERTY(texture_file, std::string,
```

Comment:
What is 0? Does it result in hiding? Is this the same as the `VisualRepresentation` above?

I have also stated before that I would prefer strings over ints since they are more informative. Any comments/response to that?